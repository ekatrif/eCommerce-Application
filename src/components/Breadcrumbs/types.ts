export type MatchingObject = {
  [key: string]: string;
};

export interface IMatch {
  match: {
    params: {
      categoryId: string;
    };
    pathname: string;
    pathnameBase: string;
    pattern: object;
    route: object;
  };
}
