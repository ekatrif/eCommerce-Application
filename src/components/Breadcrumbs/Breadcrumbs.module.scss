.list {
  display: flex;
  gap: 2em;
  margin-bottom: 1em;
  font-size: 0.9em;
  font-weight: 400;

  &>a:not(:last-child) {
    position: relative;
    width: fit-content;
  }

  &>a:not(:last-child):after {
    position: absolute;
    content: '/';
    top: 0;
    right: -1em;
  }
}