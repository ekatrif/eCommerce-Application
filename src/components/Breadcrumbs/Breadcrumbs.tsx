import useBreadcrumbs from 'use-react-router-breadcrumbs';
import { NavLink } from 'react-router-dom';
import classes from './Breadcrumbs.module.scss';
import customRoutes from '../Routing/custom-routes';

const Breadcrumbs: React.FC = () => {
  const breadcrumbs = useBreadcrumbs(customRoutes);

  return (
    <ul className={classes.list}>
      {breadcrumbs.map(({ match, breadcrumb }) => (
        <NavLink key={match.pathname} to={match.pathname}>
          <li>{breadcrumb}</li>
        </NavLink>
      ))}
    </ul>
  );
};

export default Breadcrumbs;
