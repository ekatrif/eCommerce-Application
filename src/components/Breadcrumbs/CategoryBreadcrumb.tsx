import { useSelector } from 'react-redux';
import { IRootState } from '../../features/types';
import { Languages } from '../../api/types';
import { IMatch, MatchingObject } from './types';

const CategoryBreadcrumb: React.FC<IMatch> = ({ match }) => {
  const categories = useSelector((state: IRootState) => state.categories.categoriesData);
  const categoriesNamesById: MatchingObject = categories.results.reduce((acc: MatchingObject, curr) => {
    acc[curr.id] = curr.name[Languages.English];
    return acc;
  }, {});

  return <li>{categoriesNamesById[match.params.categoryId]}</li>;
};

export default CategoryBreadcrumb;
