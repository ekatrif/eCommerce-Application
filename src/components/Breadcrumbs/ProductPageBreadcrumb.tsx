import { useSelector } from 'react-redux';
import { IRootState } from '../../features/types';
import { IMatch } from './types';

const ProductPageBreadcrumb: React.FC<IMatch> = () => {
  const productName = useSelector(
    (state: IRootState) => state.detailedProduct.detailedProductData.result?.masterData?.current?.name,
  );

  return productName ? productName['en-US'] : null;
};

export default ProductPageBreadcrumb;
