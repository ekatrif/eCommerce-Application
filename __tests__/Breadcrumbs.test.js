import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Breadcrumbs from '../src/components/Breadcrumbs/Breadcrumbs';

test('Breadcrumbs render correctly', () => {
  const tree = renderer
    .create(<MemoryRouter><Breadcrumbs /></MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
