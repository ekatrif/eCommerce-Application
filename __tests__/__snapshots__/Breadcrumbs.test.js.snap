// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs render correctly 1`] = `
<ul
  className="list"
>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <li>
      <span>
        Home
      </span>
    </li>
  </a>
</ul>
`;
